### DISCUSSION TITLE
How Can I create a queue for my music bot?

### DISCUSSION THREAD

**Discussion by ephiiPen** [Rapptz/discord.py#6971] [Q&A]
`@commands.command()
async def play(self, ctx, url):
song_there = os.path.isfile("song.mp3")
try:
if song_there:
os.remove("song.mp3")
except PermissionError:
await ctx.send("Wait for the current playing music to end or use the 'stop' command")

    voiceChannel = discord.utils.get(ctx.guild.voice_channels)
    await voiceChannel.connect()
    voice = discord.utils.get(self.client.voice_clients, guild=ctx.guild)

    ydl_opts = {
        'format': 'bestaudio/best',
        'postprocessors': [{
            'key': 'FFmpegExtractAudio',
            'preferredcodec': 'mp3',
            'preferredquality': '192',
        }]
    }
    with youtube_dl.YoutubeDL(ydl_opts) as ydl:
        ydl.download([url])
    for file in os.listdir("./"):
        if file.endswith(".mp3"):
            os.rename(file, "song.mp3")
    voice.play(discord.FFmpegPCMAudio('song.mp3'))`
    
    
    
   **This is my current code I'm using to play music through my bot, but after some searching over the internet, I am unsure of how to have the bot queue music, because whenever I try to queue two songs, it'll say in the terminal that the bot is already connected to the voice channel. I couldn't get any other help so maybe I'd be able to get some here.. Thanks!**

**Comment by tk421bsod** [comment]
discord.py doesn't handle this for you; you'll need to implement it yourself. Take a look at [this example](https://gist.github.com/EvieePy/ab667b74e9758433b3eb806c53a19f34) and [asyncio.Queue](https://docs.python.org/3/library/asyncio-queue.html).
